File: crypto.js
function raw_decrypt(ciphertext, k) {
	const key = CryptoJS.enc.Utf8.parse(k.padEnd(16, '0'));
	const decrypted = CryptoJS.AES.decrypt(ciphertext, key, {mode:CryptoJS.mode.ECB});
	return decrypted.toString(CryptoJS.enc.Utf8);
}

function raw_encrypt(plaintext, k) {
	return CryptoJS.AES.encrypt(
		plaintext,
		CryptoJS.enc.Utf8.parse(k.padEnd(16, '0')),
		{mode:CryptoJS.mode.ECB}).toString();
}

function splay(str, base, idx) {
	let res = "";

	for(let i=idx; i<str.length; i+=base) {
		res += str[i];
	}

	return res;
}

function weave(strings, base) {
	let res = "";

	for(let idx=0, pos=0; pos<strings[idx].length;) {
		res += strings[idx][pos];
		++idx;
		if(idx == base) {
			idx = 0;
			++pos;
		}
	}

	return res;
}

function decrypt(ciphertext, key) {
	let keys = Math.ceil(key.length / 16),
		decrypted = [];

	for(let i=0; i<keys; ++i) {
		decrypted.push(raw_decrypt(splay(ciphertext, keys, i), splay(key, keys, i)));
	}

	return weave(decrypted, keys);
}

let salt = "63479ad69a090b258277ec8fba6f99419a2ffb248981510657c944ccd1148e97";

function correct(key) {
	try {
		let decr = decrypt(pepper, key);
		return decr == salt;

	}catch(e){ return false; }
}

File: crypto.py
import base64 
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad,unpad

def raw_encrypt(raw, k):
    key = k.ljust(16, '0')
    raw = pad(raw.encode(),16)
    cipher = AES.new(key.encode('utf-8'), AES.MODE_ECB)
    return base64.b64encode(cipher.encrypt(raw)).decode("utf-8", "ignore")

def raw_decrypt(enc, k):
    key = k.ljust(16, '0')
    enc = base64.b64decode(enc)
    cipher = AES.new(key.encode('utf-8'), AES.MODE_ECB)
    return unpad(cipher.decrypt(enc),16).decode("utf-8", "ignore")

def splay(s, base, idx):
    res = ""

    for i in range(idx, len(s), base):
        res += s[i]

    return res

def weave(s, base):
    res = ""

    idx, pos = 0, 0
    while pos < len(s[idx]):
        res += s[idx][pos]
        idx += 1
        if idx == base:
            idx = 0
            pos += 1

    return res

def encrypt(plaintext, key):
    keys = 1 + (len(key)-1) // 16
    encrypted = []

    for i in range(keys):
        encrypted.append(raw_encrypt(splay(plaintext, keys, i), splay(key, keys, i)))

    return weave(encrypted, keys)

def decrypt(ciphertext, key):
    keys = 1 + (len(key)-1) // 16
    decrypted = []

    for i in range(keys):
        decrypted.append(raw_decrypt(splay(ciphertext, keys, i), splay(key, keys, i)))

    return weave(decrypted, keys)
